### PR TITLE
arrayjit: upper bounds to prepare for a breaking release cudajit 0.5.0

### PR DESCRIPTION
**Pull request by lukstafi**
cudajit 0.5.0 will refactor a flat interface into a modular interface.

### CODE REVIEW

File: packages/arrayjit/arrayjit.0.4.0/opam [hunk 1]
```diff
@@ -33,6 +33,7 @@ depopts: [
   "cudajit" {>= "0.4.0"}
   "gccjit" {>= "0.3.2"}
 ]
+conflicts: ["cudajit"  {>= "0.5.0"}]
```
Comment (mseri) [inline packages/arrayjit/arrayjit.0.4.0/opam:36]:
```suggestion
```

File: packages/arrayjit/arrayjit.0.4.0/opam [hunk 2]
```diff
@@ -33,6 +33,7 @@ depopts: [
   "cudajit" {>= "0.4.0"}
```
Comment (mseri) [inline packages/arrayjit/arrayjit.0.4.0/opam:33]:
```suggestion
  "cudajit" {>= "0.4.0" & < "0.5.0"}
```
Add the bound this instead, it makes it easier for the solver

File: packages/arrayjit/arrayjit.0.4.1/opam [hunk 3]
```diff
@@ -34,6 +34,7 @@ depopts: [
   "cudajit" {>= "0.4.1"}
   "gccjit" {>= "0.3.2"}
 ]
+conflicts: ["cudajit"  {>= "0.5.0"}]
```
Comment (mseri) [inline packages/arrayjit/arrayjit.0.4.1/opam:37]:
```suggestion
```

File: packages/arrayjit/arrayjit.0.4.1/opam [hunk 4]
```diff
@@ -34,6 +34,7 @@ depopts: [
   "cudajit" {>= "0.4.1"}
```
Comment (mseri) [inline packages/arrayjit/arrayjit.0.4.1/opam:34]:
```suggestion
  "cudajit" {>= "0.4.1" & < "0.5.0"}
```

**Comment by lukstafi** [comment]
Thank you for the changes! I had an outdated or misguided recollection that `depopts` does not support proper enforcement of bounds.

**Comment by mseri** [comment]
Oh no, you are perfectly right. The bound there are meaningless. I saw them as normal constraints. Let me update the comment

**Comment by raphael-proust** [comment]
LGTM. good to merge once the CI passes

**Comment by lukstafi** [comment]
Apologies re: `Merge branch`, I'm bad at git. Fixing missing maintainer email.

**Comment by shonfeder** [comment]
The windows failure is from unrelated issues with `mingw-w64-shims`.

LGTM. Thanks!

**Pull request merged**